sdk/core/Azure.Core/src/Pipeline/Internal/ResponseBodyPolicy.cs

```diff
@@ -68,7 +68,7 @@ private async ValueTask ProcessAsync(HttpMessage message, ReadOnlyMemory<HttpPip
             }
 
             Stream? responseContentStream = message.Response.ContentStream;
-            if (responseContentStream == null || responseContentStream.CanSeek)
```

Comment:
It looks like this may have been intended to optimize the case where a seekable stream is owned by the HttpResponseMessage held in e.g. [PipelineResponse](https://github.com/Azure/azure-sdk-for-net/blob/main/sdk/core/Azure.Core/src/Pipeline/HttpClientTransport.cs#L624).  If it's in-memory, we shouldn't need to buffer it.  There's a problem, however, that when the response message is [Disposed here](https://github.com/Azure/azure-sdk-for-net/blob/main/sdk/core/Azure.Core/src/Pipeline/HttpClientTransport.cs#L668), the stream it's holding gets disposed too, and it hasn't been buffered.

The question is - is the perf cost of this ok to accept?  

Are there other cases where CanSeek might be true that we would want to preserve?  Why would a response coming back be seekable unless we were using a custom transport?  If it's just a check that it's already been buffered, we could use our own subtype of memory stream to do the buffering and check for that type instead of checking CanSeek?